hackathon_site/hardware/serializers.py

```diff
@@ -89,6 +89,54 @@ def get_team_code(obj: Order):
         return obj.team.team_code
 
 
+class OrderChangeSerializer(serializers.ModelSerializer):
+    hardware = HardwareSerializer(many=True, read_only=True)
+    team_code = serializers.SerializerMethodField()
+
+    change_options = {
+        "Submitted": ["Cancelled", "Ready for Pickup"],
+        "Ready for Pickup": ["Picked Up"],
+    }
+
+    class Meta:
+        model = Order
+        fields = OrderListSerializer.Meta.fields
```

Comment:
If you're going to do this, want to just define the class as `class OrderChangeSerializer(OrderListSerializer)`, then you won't have to specify the `hardware`, `team_code`, and `get_team_code` fields / methods? Meta class stays the same

Reply:
Also, this will only work because `hardware` and `team_code` are on the list of read only fields. If eg hardware needed to be modified too by passing a different hardware ID, then you wouldn't be able to subclass because of the nested serializer.

Reply:
cool